### PR TITLE
Created abi test workflow

### LINKED ISSUE #56: Check that ABI is not changed once #54 is merged
LGTM, thanks! Could it be checked whether ABI is not changed once this PR is merged?

_Originally posted by @ni4 in https://github.com/rnpgp/sexpp/pull/54#pullrequestreview-2106116047_

### PR DESCRIPTION
**Pull request by maxirmx**

### CODE REVIEW

**Comment by maxirmx** [comment]
cc @remicollet

File: .github/workflows/build-and-test-rh.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        image: [ 'centos:7', 'quay.io/centos/centos:stream9', 'fedora:35', 'fedora:36', 'fedora:38']
+        image: [ 'centos:7', 'quay.io/centos/centos:stream9', 'fedora:36', 'fedora:38', 'fedora:40']
```
Comment (ni4) [inline .github/workflows/build-and-test-rh.yml]: Should not we leave for Fedora only v39 and v40 as the previous ones already EOLed?

**Review by ni4 — APPROVED** [review]
LGTM, thanks! Just minor CI comment.

File: .github/workflows/build-and-test-rh.yml (hunk 1 above)
Comment (maxirmx) [reply .github/workflows/build-and-test-rh.yml]: Left Fedora 39, Fedora 40

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**